modules/processors/multidomain_dataset_processor.py

```diff
@@ -444,3 +536,411 @@ def chunk_text(text, title, max_size=1000, overlap=200):
+                    break
+                chunk = text[start:end]
+                chunk = title + ": " + chunk  # Prepend the title
+                chunks.append({'id': f"{id}_{chunk_id}", 'content': chunk})
+                start = end - overlap
+                chunk_id += 1
+
+            return chunks
+        
+        plot_ids = set(dataset["plot_id"])
+        plots = {plot_id:None for plot_id in plot_ids}
+        all_chunks = []
+        for i in tqdm(range(len(dataset))):
+            if plots[dataset[i]["plot_id"]] is None:
+                id = dataset[i]["plot_id"]
+                doc = dataset[i]["plot"]
+                title = dataset[i]["title"]
+                chunks = chunk_text(doc, title, id)
+                all_chunks.extend(chunks)
+                plots[dataset[i]["plot_id"]] = True
+        dataset = datasets.Dataset.from_pandas(pd.DataFrame(all_chunks))
+        return dataset
+    
+
+# problem with this dataset: the questions are not designed for a whole datastore but rather with a fixed given context
+class CovidQA(Processor):
+    """
+    Paper: https://aclanthology.org/2020.nlpcovid19-acl.18/
+    Source: https://github.com/deepset-ai/COVID-QA
+    HF Source: https://huggingface.co/datasets/deepset/covid_qa_deepset
+    """
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'CovidQA'
+        super().__init__(*args, **kwargs, dataset_name=dataset_name)
+
+    def process(self):
+        def map_fn(row):
+            row["label"] = row["answers"]["text"]
+            return row
+        dataset = datasets.load_dataset("deepset/covid_qa_deepset")[self.split].rename_column("question","content").map(map_fn, num_proc=self.num_proc).remove_columns(["document_id", "context", "is_impossible", "answers"]).cast_column('id', datasets.Value('string'))
+        return dataset
+    
+class CORD19(Processor):
+    """
+    Paper: https://aclanthology.org/2020.nlpcovid19-acl.1.pdf
+    HF Source: https://huggingface.co/datasets/allenai/cord19
+    """
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'CORD19'
+        super().__init__(*args, **kwargs, dataset_name=dataset_name)
+
+    def process(self):
+        dataset = datasets.load_dataset("allenai/cord19", "fulltext", trust_remote_code=True)['train'] # only one split
+        all_chunks = []
+        for i in tqdm(range(len(dataset))):
+            doc = dataset[i]["fulltext"]
+            title = dataset[i]["title"]
+            chunks = chunk_text(doc, str(i), title, max_size=100, overlap=20, words_or_chars='words')
+            all_chunks.extend(chunks)
+        dataset = datasets.Dataset.from_pandas(pd.DataFrame(all_chunks))
+
+        return dataset
+    
+class LoTTE(Processor):
+    """
+    Source: https://downloads.cs.stanford.edu/nlp/data/colbert/colbertv2/lotte.tar.gz
+    Other possible source: https://huggingface.co/colbertv2
+    """
+    def __init__(self, url, tmp_download_dir='raw_datasets', *args, **kwargs):
```

Comment:
so tmp_download_dir is only used once ?
You should use 'tempfile' package instead:
```
import tempfile

# Create a temporary directory
with tempfile.TemporaryDirectory() as temp_dir:
    print(f"Temporary directory created at: {temp_dir}")
    # Use the temporary directory here

# The directory and its contents are automatically cleaned up when the block ends
```

Reply:
done, thanks for the tip 👍